src/main/kotlin/org/openrewrite/Licenses.kt

```diff
@@ -1,24 +1,29 @@
 package org.openrewrite
 
-enum class License {
-    Apache2,
-    MSAL,
-    Proprietary;
+import java.net.URI
 
-    private fun label() = when(this) {
-        Apache2 -> "Apache License Version 2.0"
-        MSAL -> "Moderne Source Available"
-        Proprietary -> "Moderne Proprietary"
-    }
-    private fun url() = when(this) {
-        Apache2 -> "https://www.apache.org/licenses/LICENSE-2.0"
-        MSAL -> "https://docs.moderne.io/licensing/moderne-source-available-license"
-        Proprietary -> "https://docs.moderne.io/licensing/overview"
+data class License (val uri: URI, val name: String) {
+    fun markdown() = "[${name}](${uri})"
+}
+
+data object Licenses {
+    val Apache2 = License(URI("https://www.apache.org/licenses/LICENSE-2.0"), "Apache License Version 2.0")
+    val MSAL = License(URI("https://docs.moderne.io/licensing/moderne-source-available-license"), "Moderne Source Available")
+    val Proprietary = License(URI("https://docs.moderne.io/licensing/overview"), "Moderne Proprietary")
+    val Unknown = License(URI(""), "License Unknown")
+
+    fun get(url: String?, name: String?): License = when {
+        url == null -> Unknown
+        url == "https://www.apache.org/licenses/LICENSE-2.0" -> Apache2
+        url == "https://docs.moderne.io/licensing/moderne-source-available-license" -> MSAL
+        url == "https://docs.moderne.io/licensing/overview" -> Proprietary
+        name != null -> License(URI(url), name)
+        else -> Unknown
     }
-    fun markdown() = "[${label()}](${url()})"
 }
 
-fun getLicense(recipeOrigin: RecipeOrigin): License {
+//todo remove logic after all recipe artifacts have been released with MANIFEST.MF containing license information
+fun getLicenseFallback(recipeOrigin: RecipeOrigin): License {
```

Comment:
Let's remove this fallback already, now that nearly all modules have been released with an updated manifest.